### PR TITLE
{CI} Fix regression test pipeline

### PR DESCRIPTION
**Pull request by evelyn-ys**
**Related command**

**Description**

After enabling MFA, regression test pipeline keeps failing with :
```
AADSTS50076: Due to a configuration change made by your administrator, or because you moved to a new location, you must use multi-factor authentication to access '797f4846-ba00-4fd7-ba43-dac1f8f63013'. Trace ID: 7108615d-16d4-4949-ab36-1cc83cfb7900 Correlation ID: 7dd265e3-980e-4065-be5b-a75b9db28359 Timestamp: 2024-01-29 18:09:48Z
Interactive authentication is needed
```
This PR removed the login with username password and use service connection instead

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
CI

File: scripts/regression_test/extension_regression_test.yml [hunk 1]
```diff
@@ -106,17 +106,20 @@ jobs:
...
-
-      source env/bin/activate
-      cd azure-cli-extensions
-
-      az login -u $(CLI_LIVE_TEST_ACCOUNT) -p "$(CLI_LIVE_TEST_PASSWORD)"
-      az account set -s 0b1f6471-1bf0-4dda-aec3-cb9272f09590
-
-      python ../scripts/ci/automation_full_test.py "12" "$(Instance_idx)" "latest" "" "True" "extension"
-    displayName: "Rerun tests"
+  - task: AzureCLI@1
```
Comment (bebound) [inline scripts/regression_test/extension_regression_test.yml:109]: Any reason to use v1 instead of v2?

Comment (evelyn-ys) [reply scripts/regression_test/extension_regression_test.yml:109]: totally fine to migrate to v2. It's just initially copied from somewhere else and v1 can already satisfy our requirements.

**Review by wangzelin007 — APPROVED** [review]

**Review by bebound — APPROVED** [review]

File: scripts/regression_test/regression_test.yml [hunk 2]
```diff
@@ -144,7 +147,7 @@ jobs:
     - task: AzureCLI@1
       displayName: 'Create PR'
       inputs:
-        azureSubscription: 'Azure CLI release'
+        azureSubscription: $(AZURE_SDK_INFRA_SUB_CONNECTED_SERVICE)
```
Comment (jiasli) [inline scripts/regression_test/regression_test.yml:150]: Don't expose `AZURE_SDK_INFRA_SUB`. Instead, consider using `REGRESSION_TEST_SERVICE_CONNECTION`.

File: scripts/regression_test/extension_regression_test.yml [hunk 3]
```diff
@@ -24,7 +24,7 @@ jobs:
     - task: AzureCLI@1
       displayName: 'checkout branch'
       inputs:
-        azureSubscription: 'Azure CLI release'
+        azureSubscription: $(AZURE_SDK_INFRA_SUB_CONNECTED_SERVICE)
```
Comment (jiasli) [inline scripts/regression_test/extension_regression_test.yml:27]: Don't expose `AZURE_SDK_INFRA_SUB`. Instead, consider using `REGRESSION_TEST_SERVICE_CONNECTION`.

File: scripts/regression_test/regression_test.yml [hunk 4]
```diff
@@ -114,21 +114,24 @@ jobs:
...
+        inlineScript: |
+          set -ev
+  
+          source env/bin/activate
+  
+          if [[ -n "$(CUSTOM_WHL_URL)" ]]; then
+            pip install $(CUSTOM_WHL_URL) --force-reinstall
+          fi
+  
+          az account set -s 0b1f6471-1bf0-4dda-aec3-cb9272f09590
```
Comment (jiasli) [inline scripts/regression_test/regression_test.yml:131]: Consider extracting subscription ID to a variable as well.

**Pull request merged**